sdks/python/apache_beam/transforms/trigger.py

```diff
@@ -1153,11 +1196,10 @@ def merge(_, to_be_merged, merge_result):  # pylint: disable=no-self-argument
         self.trigger_fn.on_element(value, window, context)
 
       # Maybe fire this window.
-      watermark = MIN_TIMESTAMP
-      if self.trigger_fn.should_fire(TimeDomain.WATERMARK, watermark,
+      if self.trigger_fn.should_fire(TimeDomain.WATERMARK, input_watermark,
                                      window, context):
-        finished = self.trigger_fn.on_fire(watermark, window, context)
-        yield self._output(window, finished, state, output_watermark, False)
+        finished = self.trigger_fn.on_fire(input_watermark, window, context)
+        yield self._output(window, finished, state, input_watermark, False)
```

Comment:
Changing this from output_watermark to input_watermark looks like it will change which panes get marked EARLY vs ON_TIME.  It's a bit worrying that this didn't cause any tests to fail

Reply:
Ack, I added both the input and output watermarks as input to the _output method.